hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/HtsSystemContract.java

```diff
@@ -50,12 +52,24 @@ public FullResult computeFully(@NonNull final Bytes input, @NonNull final Messag
 
         final HtsCall call;
         try {
-            call = attemptFactory.createCallFrom(input, frame);
-        } catch (RuntimeException ignore) {
-            // Halt and consume all remaining gas if call could not be decoded
+            call = callFactory.createCallFrom(input, frame);
+        } catch (RuntimeException e) {
+            log.debug("Failed to create HTS call from input {}", input, e);
             return haltResult(ExceptionalHaltReason.INVALID_OPERATION, frame.getRemainingGas());
         }
-        final var pricedResult = call.execute();
+
+        return resultOfExecuting(call, input, frame);
+    }
+
+    private static FullResult resultOfExecuting(
+            @NonNull final HtsCall call, @NonNull final Bytes input, @NonNull final MessageFrame frame) {
+        final HtsCall.PricedResult pricedResult;
+        try {
+            pricedResult = call.execute();
+        } catch (Exception internal) {
+            log.error("Unhandled failure for input {} to HTS system contract", input, internal);
+            return haltResult(ExceptionalHaltReason.PRECOMPILE_ERROR, frame.getRemainingGas());
+        }
         if (pricedResult.nonGasCost() > 0) {
             throw new AssertionError("Not implemented");
```

Comment:
What are non gas costs when executing system contracts?

Reply:
`TokenCreate` is so expensive that if charged purely in gas it would consume ~15M by itself. 

So instead the caller has to send hbar value directly to `0x167` when making the `tokenCreate()` call.

We then pay the fee with a combination of gas and hbar.